src/content/docs/en/recipes/dynamically-importing-images.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Dynamically Import Images
+description: Learn how to dynamically import images using Vite's import.meta.glob function
+i18nReady: true
+type: recipe
+---
+import FileTree from '~/components/FileTree.astro'
+
+When working with custom components that include images, there will be times where you will need to dynamically import the image paths. But if the images are located in the `src` directory, it is possible to run into Astro or Vite import errors.  
+
+In this recipe, you will learn how to dynamically import your images the correct way using Vite's `import.meta.glob` function. 
+
+## Recipe
+
+1. Create a new `assets` folder under the `src` directory and add your images inside that new folder. 
+
+<FileTree>
+- src/
+  - assets/
+    - example-image.jpg
+    - example-image-2.jpg
+    - example-image-3.jpg
+</FileTree>
+
+:::note
+`assets` is a popular folder name convention for placing images but you are free to name the folder whatever you like.
+::: 
+
+2. Create a new Astro component and import the `<Image>` component.
+
+    ```astro title="src/components/MyCustomCardComponent.astro" 
+    ---
+     import { Image } from 'astro:assets';
+    ---
+    ```
+
+3. Create the `props` for your component.
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={3}
+    ---
+     import { Image } from 'astro:assets';
+     const { imagePath, altText, name, age } = Astro.props;
+    ---
+    ``` 
+
+4. Create a new `images` variable and use the `import.meta.glob` function which returns an object of all of the image paths inside the `assets` folder. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={4}
+    ---
+     import { Image } from 'astro:assets';
+     const { imagePath, altText, name, age } = Astro.props;
+     const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
+    ---
+    ```
+
+5. Create the markup for your component. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={6-10}
+    ---
+     import { Image } from 'astro:assets';
+     const { imagePath, altText, developer, age } = Astro.props;
+     const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
+    ---
+    <div class="card">
+        <h2>{developer}</h2>
+        <p>Age: {age}</p>
+        <Image src={} alt={altText} />
+    </div>
+    ```
+
+6. Inside the `src` attribute, pass in the `images` object and use bracket notation for the image path. Then make sure to invoke the glob function. 
+   
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={9}
+    ---
+     import { Image } from 'astro:assets';
+     const { imagePath, altText, developer, age } = Astro.props;
+     const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
```

Comment:
The return shape of this function is a bit peculiar, it might be helpful to show it so people *understand* how it works

Reply:
I think it's `Record<string, () => Promise<{ default: ImageMetadata }>>`?